src/main/java/com/mathworks/ci/MatlabBuild.java

```diff
@@ -28,46 +30,47 @@
         //Get node specific tmp directory to copy matlab runner script
         String tmpDir = getNodeSpecificTmpFolderPath();
         FilePath targetWorkspace = new FilePath(launcher.getChannel(), tmpDir);
         ProcStarter matlabLauncher;
         if(launcher.isUnix()) {
-            matlabLauncher = launcher.launch().pwd(workspace).envs(envVars).cmds(tmpDir+"/run_matlab_command.sh",matlabCommand).stdout(listener);
+        	final String runnerScriptName = uniqueName+"/run_matlab_command.sh";
+            matlabLauncher = launcher.launch().pwd(workspace).envs(envVars).cmds(tmpDir+"/"+runnerScriptName,matlabCommand).stdout(listener);
             
             //Copy runner .sh for linux platform in workspace.
-            copyFileInWorkspace(MatlabBuilderConstants.SHELL_RUNNER_SCRIPT, "Builder.matlab.runner.script.target.file.linux.name", targetWorkspace);
+            copyFileInWorkspace(MatlabBuilderConstants.SHELL_RUNNER_SCRIPT, runnerScriptName, targetWorkspace);
         }else {
+        	final String runnerScriptName = uniqueName+"\\run_matlab_command.bat";
             launcher = launcher.decorateByPrefix("cmd.exe","/C");
-            matlabLauncher = launcher.launch().pwd(workspace).envs(envVars).cmds(tmpDir+"\\"+"run_matlab_command.bat","\""+matlabCommand+"\"").stdout(listener);
+            matlabLauncher = launcher.launch().pwd(workspace).envs(envVars).cmds(tmpDir+"\\"+runnerScriptName,"\""+matlabCommand+"\"").stdout(listener);
             //Copy runner.bat for Windows platform in workspace.
-            copyFileInWorkspace(MatlabBuilderConstants.BAT_RUNNER_SCRIPT, "Builder.matlab.runner.script.target.file.windows.name", targetWorkspace);
+            copyFileInWorkspace(MatlabBuilderConstants.BAT_RUNNER_SCRIPT, runnerScriptName, targetWorkspace);
         }
         return matlabLauncher;
     }
 
-    /**
+    /*
      * Method to copy given file from source to target node specific workspace.
      */
     default void copyFileInWorkspace(String sourceFile, String targetFile,
             FilePath targetWorkspace) throws IOException, InterruptedException {
         final ClassLoader classLoader = getClass().getClassLoader();
-        FilePath targetFilePath = new FilePath(targetWorkspace, Message.getValue(targetFile));
+        FilePath targetFilePath = new FilePath(targetWorkspace, targetFile);
         InputStream in = classLoader.getResourceAsStream(sourceFile);
         targetFilePath.copyFrom(in);
         // set executable permission
-        targetFilePath.chmod(0755);
-        
+        targetFilePath.chmod(0755);       
     }
     
-    default FilePath getNodeSpecificMatlabRunnerScript(Launcher launcher) throws IOException, InterruptedException {
+    default FilePath getNodeSpecificMatlabRunnerScript(Launcher launcher,String uniqueName) throws IOException, InterruptedException {

```

Comment:
Seems like this method name should change to reflect this is now returning a folder and not a script path.

Reply:
Done.